ports/libcopp/portfile.cmake

```
@@ -1,31 +1,36 @@
-include(vcpkg_common_functions)
-
-vcpkg_check_linkage(ONLY_STATIC_LIBRARY)
-
 vcpkg_from_github(
     OUT_SOURCE_PATH SOURCE_PATH
     REPO owt5008137/libcopp
-    REF 1.2.1
-    SHA512 bd3525b7cafb261b395ab767d30654ee7c2920f2b8312ed7887b004e764b8dd8dece5f34a5f7724d16c4a56a281ea9eb3107eff54c947160fbf9f12b76b34485
+    REF 1.3.1
+    SHA512 0401f91a98b5e3961bcda45a6249fec72f03c88ae5155f4406ba8922f47ccf187f65d6f13dedb6dcd6934bb6cb742a34e84a2fd29ca9cc5b87ca19a517a64591
     HEAD_REF v2
 )
 
 # Use libcopp's own build process, skipping examples and tests
-vcpkg_configure_cmake(
-    SOURCE_PATH ${SOURCE_PATH}
-    DISABLE_PARALLEL_CONFIGURE
-    # PREFER_NINJA # Disabled because Ninja does not invoke masm correctly for this project
-)
+if (VCPKG_LIBRARY_LINKAGE EQUAL "static")
+    vcpkg_configure_cmake(
+        SOURCE_PATH ${SOURCE_PATH}
+        DISABLE_PARALLEL_CONFIGURE
+        # PREFER_NINJA # Disabled because Ninja does not invoke masm correctly for this project
+    )
+else()
+    vcpkg_configure_cmake(
+        SOURCE_PATH ${SOURCE_PATH}
+        DISABLE_PARALLEL_CONFIGURE
+        # PREFER_NINJA # Disabled because Ninja does not invoke masm correctly for this project
+    )
+endif ()
```

```suggestion
vcpkg_configure_cmake(
    SOURCE_PATH ${SOURCE_PATH}
    DISABLE_PARALLEL_CONFIGURE
    # PREFER_NINJA # Disabled because Ninja does not invoke masm correctly for this project
)
```
The condition `if()` is not not needed. We should just keep it as before.

---

Sorr, my mistake. Fixed now.